pkg/descheduler/controllers/migration/filter.go

```diff
@@ -17,450 +17,14 @@ limitations under the License.
-	"github.com/koordinator-sh/koordinator/pkg/descheduler/framework"
-	"github.com/koordinator-sh/koordinator/pkg/descheduler/framework/plugins/kubernetes/defaultevictor"
-	podutil "github.com/koordinator-sh/koordinator/pkg/descheduler/pod"
-	pkgutil "github.com/koordinator-sh/koordinator/pkg/util"
-	utilclient "github.com/koordinator-sh/koordinator/pkg/util/client"
-)
-
-const (
-	// AnnotationPrepareMigrating is an internal switch flag, indicating that the current Pod is ready for migration,
-	// and its activation triggers certain filter logic.
-	AnnotationPrepareMigrating = "descheduler.koordinator.sh/prepare-migrating"
 )
 
-func markPodPrepareMigrating(pod *corev1.Pod) {
-	if pod.Annotations == nil {
-		pod.Annotations = make(map[string]string)
-	}
-	pod.Annotations[AnnotationPrepareMigrating] = "true"
-}
-
-func isPodPrepareMigrating(pod *corev1.Pod) bool {
-	return pod.Annotations[AnnotationPrepareMigrating] == "true"
-}
-
-func (r *Reconciler) initFilters(args *deschedulerconfig.MigrationControllerArgs, handle framework.Handle) error {
-	defaultEvictorArgs := &defaultevictor.DefaultEvictorArgs{
-		NodeFit:                 args.NodeFit,
-		NodeSelector:            args.NodeSelector,
-		EvictLocalStoragePods:   args.EvictLocalStoragePods,
-		EvictSystemCriticalPods: args.EvictSystemCriticalPods,
-		IgnorePvcPods:           args.IgnorePvcPods,
-		EvictFailedBarePods:     args.EvictFailedBarePods,
-		LabelSelector:           args.LabelSelector,
-	}
-	if args.PriorityThreshold != nil {
-		defaultEvictorArgs.PriorityThreshold = &k8sdeschedulerapi.PriorityThreshold{
-			Name:  args.PriorityThreshold.Name,
-			Value: args.PriorityThreshold.Value,
-		}
-	}
-	defaultEvictor, err := defaultevictor.New(defaultEvictorArgs, handle)
-	if err != nil {
-		return err
-	}
-
-	var includedNamespaces, excludedNamespaces sets.String
-	if args.Namespaces != nil {
-		includedNamespaces = sets.NewString(args.Namespaces.Include...)
-		excludedNamespaces = sets.NewString(args.Namespaces.Exclude...)
-	}
-
-	filterPlugin := defaultEvictor.(framework.FilterPlugin)
-	wrapFilterFuncs := podutil.WrapFilterFuncs(
-		util.FilterPodWithMaxEvictionCost,
-		filterPlugin.Filter,
-		r.filterExpectedReplicas,
-	)
-	podFilter, err := podutil.NewOptions().
-		WithFilter(wrapFilterFuncs).
-		WithNamespaces(includedNamespaces).
-		WithoutNamespaces(excludedNamespaces).
-		BuildFilterFunc()
-	if err != nil {
-		return err
-	}
-	retriablePodFilters := podutil.WrapFilterFuncs(
-		r.filterLimitedObject,
-		r.filterMaxMigratingPerNode,
-		r.filterMaxMigratingPerNamespace,
-		r.filterMaxMigratingOrUnavailablePerWorkload,
-	)
-	r.retryablePodFilter = func(pod *corev1.Pod) bool {
-		return evictionsutil.HaveEvictAnnotation(pod) || retriablePodFilters(pod)
-	}
-	r.nonRetryablePodFilter = podFilter
-	r.defaultFilterPlugin = defaultEvictor.(framework.FilterPlugin)
-	return nil
-}
-
 // Filter checks if a pod can be evicted
 func (r *Reconciler) Filter(pod *corev1.Pod) bool {
```

Comment:
We can move the Filter and PreFilter methods to controller.go.

Reply:
yes